src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java

```diff
@@ -596,18 +596,34 @@ public Boolean apply(WebDriver driver) {
   }
 
   public AdsBaseObject triggerFloatingMedrec() {
+    final String floatingMedrecCssSelector = AdsContent.getSlotSelector(AdsContent.FLOATING_MEDREC);
     try {
       new WebDriverWait(driver, 5).until(new ExpectedCondition<Object>() {
         @Override
         public Object apply(WebDriver webDriver) {
           jsActions.execute(
               " (function(){ window.scroll(0, 5000); setTimeout(function () {window.scroll(0, 5001) }, 100); })(); ");
-          String floatingMedrecCssSelector = AdsContent.getSlotSelector(AdsContent.FLOATING_MEDREC);
           return driver.findElements(By.cssSelector(floatingMedrecCssSelector)).size() > 0;
         }
       });
     } catch (org.openqa.selenium.TimeoutException e) {
-      PageObjectLogging.logError("Floating Medrec", e);
+      PageObjectLogging.logError("Floating medrec", e);
+    }
+    return this;
+  }
+
+  public AdsBaseObject triggerIncontentLeaderboard() {
```

Comment:
Looks like quite a lot of code duplication. Can we think of something more generic and refactor our tests a little bit? Something like `public AdsBase Object triggerAdSlot(String adSlotSelector, String javaScriptTrigger)`...